### PR TITLE
Added theming on XVimStatusLine, fixed clipping bug with DVTSourceTex…

### PR DESCRIPTION
**Pull request by richardjrossiii**
…tView when selecting code blocks in gutter

Screenshot of the issue:

screen shot 2015-05-26 at 4 44 47 pm

### CODE REVIEW

File: XVim/DVTSourceTextView+XVim.m [hunk 1]
```diff
@@ -165,9 +165,9 @@ - (void)xvim_drawRect:(NSRect)dirtyRect{
         }else{
             [self xvim_clearHighlightText];
         }
-        
-        [self xvim_drawRect:dirtyRect];
-        
+
+        [self xvim_drawRect:[self visibleRect]];
```
Comment (richardjrossiii) [inline XVim/DVTSourceTextView+XVim.m]: Using dirtyRect here causes our gutter highlighting to be completely broken. VisibleRect fixes this with no noticeable performance decrease. In my tests.

**Comment by rokemoon** [comment]
I get regression with this fix http://monosnap.com/image/i26Nb9swV5naUawYVIjZUuOwtuc2b3.
Debug line with green background are disappeared, if you set mouse on this line it appears and then disappeared again.

**Comment by richardjrossiii** [comment]
This refactors the caret drawing code, but it works now with annotations properly.
screen shot 2015-05-27 at 1 21 51 pm

**Comment by richardjrossiii** [comment]
Any status updates on this PR?

**Comment by JugglerShu** [comment]
I have merged this change into develop branch. I'll see if any problem occurs in the branch.

**Comment by JugglerShu** [comment]
Thank you for the report. The cursor part is complex and not easy to achieve right behavior.
I might just remove this merge and see what I can do to fix the original issue.

**Pull request closed without merging**